pkg/server/plugin/nodeattestor/k8s/sat.go

```diff
@@ -0,0 +1,307 @@
+		Valid:        true,
+		BaseSPIFFEID: k8s.AgentID(config.trustDomain, attestationData.UUID),
+		Selectors: []*common.Selector{
+			makeSelector("cluster:name", cluster.name),
+			makeSelector("service-account:namespace", claims.Namespace),
+			makeSelector("service-account:name", claims.ServiceAccountName),
+		},
+	})
+}
+
+func (p *SATAttestorPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
+	hclConfig := new(SATAttestorConfig)
+	if err := hcl.Decode(hclConfig, req.Configuration); err != nil {
+		return nil, satError.New("unable to decode configuration: %v", err)
+	}
+	if req.GlobalConfig == nil {
+		return nil, satError.New("global configuration is required")
+	}
+	if req.GlobalConfig.TrustDomain == "" {
+		return nil, satError.New("global configuration missing trust domain")
+	}
+
+	if len(hclConfig.Clusters) == 0 {
+		return nil, satError.New("configuration must have at least one cluster")
+	}
+
+	config := &satAttestorConfig{
+		trustDomain: req.GlobalConfig.TrustDomain,
+	}
+	config.trustDomain = req.GlobalConfig.TrustDomain
+	for name, cluster := range hclConfig.Clusters {
+		if cluster.ServiceAccountKeyFile == "" {
+			return nil, satError.New("cluster %q configuration missing service account key file", name)
+		}
+		if len(cluster.ServiceAccountWhitelist) == 0 {
+			return nil, satError.New("cluster %q configuration must have at least one service account whitelisted", name)
+		}
+
+		serviceAccountKeys, err := loadServiceAccountKeys(cluster.ServiceAccountKeyFile)
+		if err != nil {
+			return nil, satError.New("failed to load cluster %q service account keys from %q: %v", name, cluster.ServiceAccountKeyFile, err)
+		}
+
+		if len(serviceAccountKeys) == 0 {
+			return nil, satError.New("cluster %q has no service account keys in %q", name, cluster.ServiceAccountKeyFile)
+		}
+
+		serviceAccounts := make(map[string]bool)
+		for _, serviceAccount := range cluster.ServiceAccountWhitelist {
+			serviceAccounts[serviceAccount] = true
+		}
+
+		config.clusters = append(config.clusters, &clusterConfig{
+			name:               name,
+			serviceAccountKeys: serviceAccountKeys,
+			serviceAccounts:    serviceAccounts,
+		})
+	}
+
+	p.setConfig(config)
+	return &spi.ConfigureResponse{}, nil
+}
+
+func (p *SATAttestorPlugin) GetPluginInfo(context.Context, *spi.GetPluginInfoRequest) (*spi.GetPluginInfoResponse, error) {
+	return &spi.GetPluginInfoResponse{}, nil
+}
+
+func (p *SATAttestorPlugin) getConfig() (*satAttestorConfig, error) {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	if p.config == nil {
+		return nil, satError.New("not configured")
+	}
+	return p.config, nil
+}
+
+func (p *SATAttestorPlugin) setConfig(config *satAttestorConfig) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+	p.config = config
+}
+
+func verifyTokenSignature(clusters []*clusterConfig, token *jwt.JSONWebToken) (cluster *clusterConfig, claims *k8s.SATClaims, err error) {
+	var lastErr error
+	for _, cluster := range clusters {
```

Comment:
if we make users configure the cluster name on the agent attestor, maybe we could avoid this bit

Reply:
👍 good call